libraries/management/commands/import_library_versions.py

```diff
@@ -0,0 +1,137 @@
+                        libraries.append(data)
+                else:
+                    libraries.append(parser.parse_libraries_json(libraries_json))
+
+                for lib_data in libraries:
+                    try:
+                        library = Library.objects.get(name=lib_data["name"])
+                    except Library.DoesNotExist:
+                        click.echo(
+                            f"Could not find library by gitmodule name; skipping {gitmodule['module']}"
+                        )
+                        skipped.append(
+                            {
+                                "version": version.name,
+                                "library": gitmodule["module"],
+                                "reason": "Could not find library by gitmodule name",
+                            }
+                        )
+                    else:
+                        library_version, _ = LibraryVersion.objects.get_or_create(
+                            version=version, library=library
+                        )
+                        click.echo(
+                            f"Saved library version {library_version}. Created? {_}"
+                        )
+                        updater.update_maintainers(
+                            library_version, maintainers=lib_data["maintainers"]
+                        )
+                        click.secho(
+                            f"Updated maintainers for {library_version}.", fg="green"
+                        )
+
+                    click.echo(f"Saved library version {library_version}.")
+            else:
+                # This can happen with older tags; the libraries.json file didn't always exist, so
+                # when it isn't present, we search for the library by the module name and try to save
+                # the LibraryVersion that way.
+                click.echo(
+                    f"Could not get libraries.json for {gitmodule['module']}; will try to save by gitmodule name."
+                )
+                try:
+                    library = Library.objects.get(name=gitmodule["module"])
+                except Library.DoesNotExist:
+                    click.echo(
+                        f"Could not find library by gitmodule name; skipping {gitmodule['module']}"
+                    )
+                    skipped.append(
+                        {
+                            "version": version.name,
+                            "library": gitmodule["module"],
+                            "reason": "Could not find library in database by gitmodule name",
+                        }
+                    )
+                else:
+                    library_version, _ = LibraryVersion.objects.get_or_create(
+                        version=version, library=library
+                    )
+                    click.echo(f"Saved library version {library_version}. Created? {_}")
+
+    if skipped:
+        for skipped_obj in skipped:
```

Comment:
One trick I usually do for avoiding the `if` and the extra indentation level is:

```python
for skipped_obj in skipped or []:
    do the for bits
```

Or considering we control `skipped` and we ensure across the function that it's always a list, we can just execute the `for` without any `if` since `skipped` will always be a list.

Reply:
Ooh thanks! I'm so bad about over-nesting but the little tricks like that don't occur to me.